prov/rxm/src/rxm_ep.c

```diff
@@ -1407,15 +1358,18 @@ rxm_ep_inject_send_common(struct rxm_ep *rxm_ep, const struct iovec *iov, size_t
 	int ret;
 
 	if (rxm_ep->util_ep.domain->threading != FI_THREAD_SAFE) {
+		uint64_t saved_flags = inject_pkt->hdr.flags;
```

Comment:
Why is this needed?

Reply:
This is needed to reset flags that can be overwritten in case of `fi_sendmsg`/`fi_tsendmsg` when we pass `inject_data_pkt`/`tinject_data_pkt` to `rxm_ep_send_common()`.
Probably, we need to simplify it somehow, because this is not clear enough.

Reply:
But this function calls only `rxm_ep_msg_inject_send` which doesn't seem to override the flags.

Reply:
`inject_pkt->hdr.flags = (flags & FI_REMOTE_CQ_DATA);` this modifies `flags` field of `inject_data_pkt`/`tinject_data_pkt` in case of `(flags & FI_REMOTE_CQ_DATA) == 0`

Reply:
Oops, this is just below. I guess I'm getting review fatigue.

Reply:
Perhaps we need to move the check to the caller (sendmsg, tsendmsg). Check if FI_REMOTE_CQ_DATA is set in flags. If so pass inject_data_pkt.